src/plugins/workspace/public/components/workspace_form/fields/workspace_name_field.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiCompressedFieldText, EuiCompressedFormRow, EuiTextColor } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import React, { useCallback } from 'react';
+
+import { MAX_WORKSPACE_NAME_LENGTH } from '../../../../common/constants';
+
+export interface WorkspaceNameFieldProps {
+  value?: string;
+  onChange: (newValue: string) => void;
+  error?: string;
+  readOnly?: boolean;
+}
+
+export const WorkspaceNameField = ({
+  value,
+  error,
+  readOnly,
+  onChange,
+}: WorkspaceNameFieldProps) => {
+  const handleChange = useCallback(
+    (e) => {
+      onChange(e.currentTarget.value);
+    },
+    [onChange]
+  );
+  const leftCharacters = MAX_WORKSPACE_NAME_LENGTH - (value?.length ?? 0);
+  const charactersOverflow = leftCharacters < 0;
+
+  return (
+    <EuiCompressedFormRow
+      label={i18n.translate('workspace.form.workspaceDetails.name.label', {
+        defaultMessage: 'Name',
+      })}
+      helpText={
+        <>
+          <EuiTextColor color={charactersOverflow ? 'danger' : 'subdued'}>
+            {i18n.translate('workspace.form.name.charactersLeft', {
+              defaultMessage: '{leftCharacters} characters left.',
+              values: {
+                leftCharacters,
+              },
+            })}
+          </EuiTextColor>
+          <br />
+          {i18n.translate('workspace.form.workspaceDetails.name.helpText', {
+            defaultMessage:
+              'Use a unique name for the workspace. Valid characters are a-z, A-Z, 0-9, (), [], _ (underscore), - (hyphen) and (space).',
+          })}
+        </>
+      }
+      isInvalid={!!error || charactersOverflow}
```

Comment:
So there will be a case that user inputs too many characters and the input will become Invalid but without any error message, is that expected?

Reply:
It will show `-xx characters left` and turned red. The create workspace button will be disabled too.

Reply:
I see, thanks.